### DISCUSSION TITLE
Prevent generation of db.go, when you want to create your own

### DISCUSSION THREAD

**Discussion by robfordww** [sqlc-dev/sqlc#1399] [Q&A]
While combining some custom queries with sqlc genereted ones, I have found it problematic to prevent it from overwriting the` db.go `file every time the` sqlc genereate `is run.  More specifically, I would like to create my own `Queries interface`, with expanded method set, and let it remain untouched, so that my own queries could use those methods.  I tried to set `output_db_file_name:""` , but that did not seem to work,  Is there a nice way to do this?

**Comment by robfordww** [comment]
... or is there some way to alter the templates without recompiling the library?  Could be useful.

**Reply by robfordww** [reply]
I see this can be done by adding another file to the package with a NewExtended() (or similar name) function, that returns a structure wrapping the existing Queries interface. Not sure if there are any better ways to do this...

**Comment by Jille** [comment]
I don't use the Querier interface, but only the Queries struct. That allows me to add more files in the same directory that add more methods to the Queries struct.

**Reply by skabbes** [reply]
This is a great answer - I think this is the way to go! If you want the flexibility of an interface as well.  You can even use interface embedding in Golang to extend what has been generated by sqlc.
